qiskit/ignis/verification/tomography/fitters/gateset_fitter.py

```diff
@@ -0,0 +1,662 @@
+                probs: The probabilities obtained experimentally
+                qubits: the size of the gates in the gateset
+        """
+        self.probs = probs
+        self.Gs = Gs
+        self.Fs = Fs
+        self.Fs_names = list(Fs.keys())
+        self.qubits = qubits
+        self.obj_fn_data = self.compute_objective_function_data()
+        self.initial_value = None
+
+    def compute_objective_function_data(self) -> List:
+        """Computes auxiliary data needed for efficient computation
+        of the objective function.
+
+            Returns:
+                 The objective function data list
+            Additional information:
+                The objective function is
+                sum_{ijk}(<|E*R_Fi*G_k*R_Fj*Rho|>-m_{ijk})^2
+                We expand R_Fi*G_k*R_Fj to a sequence of G-gates and store
+                indices. We also obtain the m_{ijk} value from the probs list
+                all that remains when computing the function is thus
+                performing the matrix multiplications and remaining algebra.
+        """
+        m = len(self.Fs)
+        n = len(self.Gs)
+        obj_fn_data = []
+        for (i, j) in itertools.product(range(m), repeat=2):
+            for k in range(n):
+                Fi = self.Fs_names[i]
+                Fj = self.Fs_names[j]
+                m_ijk = (self.probs[(Fi, self.Gs[k], Fj)])
+                Fi_matrices = [self.Gs.index(gate) for gate in self.Fs[Fi]]
+                Fj_matrices = [self.Gs.index(gate) for gate in self.Fs[Fj]]
+                matrices = Fi_matrices + [k] + Fj_matrices
+                obj_fn_data.append((matrices, m_ijk))
+        return obj_fn_data
+
+    def split_input_vector(self, x: np.array) -> Tuple:
+        """Reconstruct the GST data from its vector representation
+        Args:
+            x: The vector representation of the GST data
+
+        Returns:
+            The GST data (E, rho, Gs) (see additional info)
+
+        Additional information:
+            The gate set tomography data is a tuple (E, rho, Gs) consisting of
+            1) A POVM measurement operator E
+            2) An initial quantum state rho
+            3) A list Gs = (G1, G2, ..., Gk) of gates, represented as matrices
+
+            This function reconstructs (E, rho, Gs) from the vector x
+            Since the MLE optimization procedure has PSD constraints on
+            E, rho and the Choi represetnation of the PTM of the Gs,
+            we rely on the following property: M is PSD iff there exists
+            T such that M = T @ T^{dagger}.
+            Hence, x stores those T matrices for E, rho and the Gs
+        """
+        n = len(self.Gs)
+        d = (2 ** self.qubits)
+        ds = d ** 2  # d squared - the dimension of the density operator
+
+        d_t = complex_matrix_var_nums(d)
+        ds_t = complex_matrix_var_nums(ds)
+        T_vars = split_list(x, [d_t, d_t] + [ds_t] * n)
```

Comment:
I think this is the only location where split list is needed, so maybe the function can be moved to a helper function in this class, or just have its body inserted into this function.

Reply:
Done in 1477ef1